com.flashforge.FlashPrint.json

```diff
@@ -0,0 +1,81 @@
+{
+  "app-id": "com.flashforge.FlashPrint",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.9",
+  "sdk": "org.kde.Sdk",
+  "command": "FlashPrint",
+  "tags": [
+    "proprietary"
+  ],
+  "finish-args": [
+    "--filesystem=home",
+    "--socket=wayland",
+    "--socket=x11",
+    "--device=dri",
+    "--share=ipc"
+  ],
+  "cleanup": [
+    "/include",
+    "/lib/pkgconfig",
+    "/share/pkgconfig"
+  ],
+  "modules": [
+    {
+      "name": "FlashPrint",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -D apply_extra /app/bin/apply_extra",
+        "install -D flashprint-bin /app/bin/FlashPrint",
+        "install -Dm644 com.flashforge.FlashPrint.appdata.xml /app/share/appdata/com.flashforge.FlashPrint.appdata.xml",
+        "install -D /usr/bin/ar /app/bin/ar",
+        "mkdir -p /app/lib",
+        "ARCH_TRIPLE=$(gcc --print-multiarch) && cp /usr/lib/${ARCH_TRIPLE}/libbfd-*.so /app/lib"
+      ],
+      "sources": [
+        {
+          "type": "script",
+          "dest-filename": "apply_extra",
+          "commands": [
+            "ar p flashprint.deb data.tar.xz | tar xJ",
+            "rm -f flashprint.deb",
+            "mv usr/* .",
+            "rmdir usr",
+            "mkdir -p export/share/applications",
+            "sed s/Icon=flashforge/Icon=com.flashforge.FlashPrint share/applications/FlashPrint.desktop > export/share/applications/com.flashforge.FlashPrint",
+            "install -Dm644 share/icons/hicolor/64x64/apps/flashforge.png export/share/icons/hicolor/64x64/apps/com.flashforge.FlashPrint.png"
+          ]
+        },
+        {
+          "type": "extra-data",
+          "filename": "flashprint.deb",
+          "only-arches": [
+            "x86_64"
+          ],
+          "url": "http://www.sz3dp.com/upfile/2018/12/03/20181203162713_662.deb",
+          "sha256": "2bb003b7bb552dd6b04495a084d1bf03de620bb94dea44f3b6909516f4a95dc1",
+          "size": 12717922
+        },
+        {
+          "type": "file",
+          "path": "com.flashforge.FlashPrint.appdata.xml"
+        },
+        {
+          "type": "file",
+          "path": "flashprint-bin"
+        }
+      ]
+    },
+    "shared-modules/glu/glu-9.0.0.json",
+    {
+      "name": "libudev1",
```

Comment:
We also have udev in shared-modules but I'm not sure about provided soname version.

Reply:
FlashPrint requires libudev.so.1, and the shared modules provide libudev.so.0. I asked for recommendation [in a reddit thread][1] and I was advised to use [eudev][2].

[1]: https://www.reddit.com/r/Fedora/comments/ams0xd/making_a_flatpak_for_flashprint_issue_with/
[2]: https://wiki.gentoo.org/wiki/Eudev